### DISCUSSION TITLE
Major GC slice counters in runtime events

### DISCUSSION THREAD

**Discussion by kayceesrk** [ocaml/ocaml#13321] [Show and tell]
While debugging #13300, I found it useful to expose counters used in major slice work calculation through runtime events. Specifically, these counters:

https://github.com/ocaml/ocaml/blob/84335d2a9e78e0101518121914517764c0986ce1/runtime/major_gc.c#L699-L708

With these counters, one is able to clearly see in the program trace why a particular major slice was triggered (with the help of `olly`: https://github.com/tarides/runtime_events_tools/pull/46). 

![image](https://github.com/user-attachments/assets/0be95ae2-2084-4710-8cec-04778cd3c3bc)

In the above, the major slice is triggered when there is a spike in allocation (`allocated_words`, `allocation_counter`). Since the work done by GC (`work_counter`) lags behind, a slice gets triggered after which the work_counter catches up.

Would it make sense to upstream them?

These counters are emitted fairly infrequently, roughly proportional to major slices. So the cost of them is negligible.

**Comment by gasche** [comment]
Sure, I would say that anything that is passed to the `caml_gc_log` or `caml_gc_message` is fair game for runtime-events, and in fact probably a good idea because the logs/messages are there because people already found them useful to reason about the program in the past.

(Conversely: I wouldn't mind seeing `caml_gc_log`/`caml_gc_message` systematically where we have runtime events as well, because logs/messages provide a trivial, low-tech way to look for stuff -- set the right OCAMLRUNPARAM settings and then grep the result. Maybe we could just ensure that the `caml_ev_*` functions also log stuff.)

**Reply by kayceesrk** [reply]
While adding an event where there is already a `caml_gc_log` or a `caml_gc_message` sounds reasonable, we'll need to see whether the other way around makes the logs noisy. 

I'll make a PR after we see a bit more discussion on #13320 as both will touch the same code.
